lib/hello_test_server.ex

```diff
@@ -41,13 +41,39 @@ defmodule HelloTestServer do
 
   def handle_request(_context, "Server.ping", args, state) do
     case args do
-    %{"rsleep" => rsleep} -> rsleep |> to_integer() |> random() |> :timer.sleep
-    %{"sleep" => sleep}   -> sleep  |> to_integer() |> :timer.sleep
-    _ -> nil
+      %{"rsleep" => rsleep} -> rsleep |> to_integer() |> random() |> :timer.sleep
+      %{"sleep" => sleep}   -> sleep  |> to_integer() |> :timer.sleep
+      _ -> nil
     end
     {:stop, :normal, {:ok, "pong"}, state}
   end
 
+  def handle_request(_context, method, args, state) do
+    dirname = Application.app_dir(:hello_test_server, "priv/replies") |> :filename.join(method)
+    case File.ls(dirname) do
+      {:ok, files} -> 
+        counter = case :ets.lookup(@rrtable, method) do
+          [{method, n}] -> n
+          _ -> :ets.insert(@rrtable, {method, 0})
+          0
+        end
+        {ok, reply} = filter_json(dirname, files)
+                      |> choose_reply(counter)
+                      |> File.read
```

Comment:
You can use `File.read!` instead, it is a good style to use `!` because of readable exceptions,